### PR TITLE
Fix estimate exit and redeem

### PR DESCRIPTION
**Pull request by mijovic**
`getSwapAmountToEndWithEqualShares` just moved to TempusAMM.

### CODE REVIEW

File: contracts/stats/Stats.sol [hunk 1]
```diff
@@ -137,20 +137,40 @@ contract Stats is ITokenPairPriceFeed, ChainlinkTokenPairPriceFeed {
...
+
+            if (yieldsIn) {
+                principals += amountOut;
+                yields -= amountIn;
+            } else {
+                principals -= amountIn;
+                yields += amountOut;
+            }
+
+            if (principals > yields) {
```
Comment (RedFox20) [inline contracts/stats/Stats.sol:167]: This needs a short comment

File: contracts/stats/Stats.sol [hunk 2]
```diff
@@ -137,20 +137,40 @@ contract Stats is ITokenPairPriceFeed, ChainlinkTokenPairPriceFeed {
...
-            );
+        principals += principalsFromLP;
+        yields += yieldsFromLp;
+
+        if (!tempusAMM.tempusPool().matured()) {
+            bool yieldsIn = yields > principals;
+            uint256 amountIn = tempusAMM.getSwapAmountToEndWithEqualShares(principals, yields, threshold);
+            uint256 amountOut = (amountIn != 0) ? tempusAMM.getExpectedReturnGivenIn(amountIn, yieldsIn) : 0;
+
+            if (yieldsIn) {
```
Comment (RedFox20) [inline contracts/stats/Stats.sol:159]: This needs a short comment

Comment (mijovic) [reply contracts/stats/Stats.sol:160]: Added

**Comment by mijovic** [comment]
Also added test for early call to estimateExitAndRedeem

**Review by RedFox20 — APPROVED** [review]

**Pull request merged**